app/src/main.rs

```diff
@@ -45,43 +44,36 @@ fn main(_argc: isize, _argv: *const *const u8) -> isize {
         target_tests::run_test_suite()
     } else {
         println!("Starting game!");
-        main_game()
+        register_power_callback();
+        main_game();
+        shutdown();
+        0
     }
 }
 
 fn main_game() -> isize {
-    println!("Hello Rust!");
-    register_power_callback();
-
-    // Setup the wiimote
-    Input::init(ControllerType::Wii);
-    let wii_mote = Input::new(ControllerType::Wii, ControllerPort::One);
-    wii_mote
-        .as_wpad()
-        .set_data_format(WPadDataFormat::ButtonsAccelIR);
-
-    // Setup the ECS environment.
-    let mut world = World::new();
-    batch_spawn_entities(&mut world, 5);
-    let mut velocity_query = PreparedQuery::<&mut Velocity>::default();
-    let mut all_query = PreparedQuery::<(&mut Position, &mut Velocity)>::default();
-
-    // Kickstart main loop.
+    // register_power_callback();
+    let mut game_state = GameState::new();
+    let mut input_manager = InputManager::new();
     let renderer = Renderer::new();
+
     while KEEP_RUNNING.load(Ordering::SeqCst) {
-        Input::update(ControllerType::Wii);
-        if wii_mote.is_button_down(Button::Home) {
+        let controls = input_manager.update();
+
+        let changes = Changes {
+            controls,
+            delta_time_ms: 100,
```

Comment:
I assume this is due to us not being able to get the real delta time?

Reply:
Correct! I included it as an example of other data we want to pass along between the outside world and the 'pure' world inside.